igc_lib.py

```diff
@@ -71,7 +74,148 @@ def _rawtime_float_to_hms(timef):
+            else:
+                if point == tpoints[-1]:   # if it is the last turnpoint i.e. the goal
+                    if point.getAttribute("type") == "line":
+                        sort = "goal_cylinder"     # to change one line can be processed.
+                    else:
+                        sort = "goal_cylinder"
+                else:
+                    sort = "cylinder"
+            
+                      
+            turnpoint = Turnpoint(lat, lon, radius, sort)
+            turnpoints.append(turnpoint)
+        task = Task(turnpoints, start_time)
+        return task
+    
+    
+    def __init__(self, turnpoints, start_time, end_time=86399):  #finish_time defaults to 23:59
+        self.turnpoints = turnpoints
+        self.start_time = start_time
+        self.end_time = end_time
+        
+        
+        
+    def check_flight(self, flight):
+        """ Checks a flight object against the task. 
+            Args:
+                   flight: a flight object
+            Returns:
+                    a list of rawtimes of when turnpoints were achieved.
+            
+        """
+        turnpoint_times = []   
+    
+        proceed_to_start = False
+        t=0
+        
+        for fix in flight.fixes:
+        
+            
+                if self.turnpoints[t].sort == "start_exit": #pilot must have at least 1 fix inside the start after the start time then exit
+                    if proceed_to_start:
+                        if not self.turnpoints[t].in_radius(fix):
+                            turnpoint_times.append(fix.rawtime)  #pilot has started
+                            t += 1
+                    if fix.rawtime > self.start_time and not proceed_to_start:
+                        if self.turnpoints[t].in_radius(fix):
+                            proceed_to_start = True         #pilot is inside start after the start time.
+                        
+                if self.turnpoints[t].sort == "start_enter":  #pilot must have at least 1 fix outside the start after the start time then enter
+                    if proceed_to_start:
+                        if self.turnpoints[t].in_radius(fix):
+                            turnpoint_times.append(fix.rawtime)  #pilot has started
+                            t += 1
+                    if fix.rawtime > self.start_time and not proceed_to_start:   
+                        if not self.turnpoints[t].in_radius(fix):
+                            proceed_to_start = True         #pilot is outside start after the start time.    
+            
+                if self.turnpoints[t].sort in ["cylinder", "ESS", "goal_cylinder"]:
+                    if self.turnpoints[t].in_radius(fix):
+                            turnpoint_times.append(fix.rawtime)  #pilot has achieved turnpoint
+                            t += 1
+                                                  
+                            if t >= len(self.turnpoints):
```

Comment:
Remove two levels of indentation and move to the beginning of the loop - we should exit as soon as there are no more turnpoints.